auslib/db.py

```diff
@@ -1074,7 +1083,12 @@ def enactChange(self, sc_id, enacted_by, transaction=None):
 
         # If the scheduled change had a data version, it means the row already
         # exists, and we need to use update() to enact it.
-        if what["data_version"]:
+        if change_type == "delete":
```

Comment:
The _checkBaseTablePermissions block needs updating too: https://github.com/mozilla/balrog/blob/37b499a7d0c3d117186e3341eec7e2ea8f6e972d/auslib/db.py#L897.